containers/src/Kokkos_DynamicView.hpp

```diff
@@ -604,59 +604,112 @@ struct MirrorDynamicViewType {
-      Kokkos::view_alloc(arg_prop...,
-                         std::string(src.label()).append("_mirror")),
-      src.chunk_size(), src.chunk_max() * src.chunk_size());
+      prop_copy, src.chunk_size(), src.chunk_max() * src.chunk_size());
 }
 
-template <class Space, class T, class... P, class... I>
+template <class Space, class T, class... P, class... ViewCtorArgs>
 inline typename Kokkos::Impl::MirrorDynamicViewType<Space, T, P...>::view_type
 create_mirror(const Space&,
               const Kokkos::Experimental::DynamicView<T, P...>& src,
-              const I&... arg_prop) {
+              const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop) {
+  using alloc_prop_input = Impl::ViewCtorProp<ViewCtorArgs...>;
+
+  static_assert(
+      !alloc_prop_input::has_label,
+      "The view constructor arguments passed to Kokkos::create_mirror "
+      "must not include a label!");
+  static_assert(
+      !alloc_prop_input::has_pointer,
+      "The view constructor arguments passed to Kokkos::create_mirror must "
+      "not include a pointer!");
+  static_assert(
+      !alloc_prop_input::has_memory_space,
+      "The view constructor arguments passed to Kokkos::create_mirror must "
+      "not include a memory space instance!");
+  static_assert(
+      !alloc_prop_input::allow_padding,
+      "The view constructor arguments passed to Kokkos::create_mirror must "
+      "not explicitly allow padding!");
+
+  using MemorySpace = typename Space::memory_space;
+  using alloc_prop =
+      Impl::ViewCtorProp<ViewCtorArgs..., std::string, MemorySpace>;
+  alloc_prop prop_copy(arg_prop);
+  static_cast<Impl::ViewCtorProp<void, std::string>&>(prop_copy).value =
+      std::string(src.label()).append("_mirror");
+
   return
       typename Kokkos::Impl::MirrorDynamicViewType<Space, T, P...>::view_type(
-          Kokkos::view_alloc(arg_prop...,
-                             std::string(src.label()).append("_mirror")),
-          src.chunk_size(), src.chunk_max() * src.chunk_size());
+          prop_copy, src.chunk_size(), src.chunk_max() * src.chunk_size());
 }
 }  // namespace Impl
 
 // Create a mirror in host space
 template <class T, class... P>
 inline auto create_mirror(
     const Kokkos::Experimental::DynamicView<T, P...>& src) {
-  return Impl::create_mirror(src);
+  return Impl::create_mirror(src, Impl::ViewCtorProp<>{});
 }
 
 template <class T, class... P>
 inline auto create_mirror(
     Kokkos::Impl::WithoutInitializing_t wi,
     const Kokkos::Experimental::DynamicView<T, P...>& src) {
-  return Impl::create_mirror(src, wi);
+  return Impl::create_mirror(src, Kokkos::view_alloc(wi));
 }
 
 // Create a mirror in a new space
 template <class Space, class T, class... P>
 inline auto create_mirror(
     const Space& space, const Kokkos::Experimental::DynamicView<T, P...>& src) {
-  return Impl::create_mirror(space, src);
+  return Impl::create_mirror(space, src, Impl::ViewCtorProp<>{});
```

Comment:
should we just pass `view_alloc(space)` in here?

Reply:
3f5c6c43aafa38aac4c9540d47af14ecd712c2ff